src/mminou.h

```diff
@@ -93,29 +102,270 @@ extern flag g_commandFileSilentFlag; /* For SUBMIT ... /SILENT */
+ *     to the outer, temporarily suspended call.
+ * 4. __Redirection__.  Instead of displaying output on the virtual text
+ *     display, it may be stored in a NUL-terminated string variable.  Or even
+ *     completely suppressed, for example when executing a SUBMIT.
+ * 5. __Logging__.  Output may be logged to a file for examination.
+ *
+ * These effects need not be carried out in this order, some may even be
+ * omitted.  We show the order of steps and their respective conditions here.
+ *
+ * -# The \ref backBuffer is almost private to this function, so its
+ *     initialization is done here, right at the beginning.  The
+ *     \ref backBufferPos is always at least 1, so a value of 0 indicates an
+ *     outstanding \ref backBuffer memory allocation, and an empty string is
+ *     pushed as a first (guard) page onto it (see \ref pgBackBuffer).
+ *
+ *     \ref g_pntrTempAllocStackTop may be reset to
+ *     \ref g_pntrStartTempAllocStack.
+ *
+ * -# If the current page is full and further output would overflow it, as
+ *     indicated by \ref printedLines, output may be suspended as described ins
+ *     (2) and the user is prompted for scrolling actions.\n
+ *     This step is unconditionally executed when \ref backFromCmdInput = 1
+ *     (\ref cmdInput explicitely requested it).  The values in
+ *     \ref g_quitPrint and \ref localScrollMode are retained then, regardless
+ *     of user input.
+ *
+ *     Other conditions prevent this step:  Output is discarded on user request
+ *     (\ref g_quitPrint = 1), a SUBMIT call is running
+ *     (\ref g_commandFileNestingLevel > 0), scrolling is generally
+ *     (\ref g_scrollMode = 0) or temporarily (\ref localScrollMode = 0)
+ *     disabled, output is redirected (\ref g_outputToString = 1).
+ *
+ *     This step can set \ref g_quitPrint to 1 (if \ref backFromCmdInput = 0).
+ *     It can modify \ref backBufferPos (by command _q_ or _Q_) and set
+ *     \ref localScrollMode = 0 (by command _s_ or _S_, and
+ *     \ref backFromCmdInput = 0).
+ *
+ * -# If pending output would overflow the screen, \ref backBuffer is extended
+ *     by a new page to receive pending output.
+ *     Several conditions can prevent this step:  Step (2) was not executed,
+ *     output is discarded on user request (\ref g_quitPrint = 1),
+ *     \ref backFromCmdInput = 1 (\ref cmdInput needs the scrolling loop
+ *     only).
+ *
+ *     Sets \ref printedLines to 0, indicating that nothing has been added to
+ *     the new page yet. Increments \ref backBufferPos so all pending output is
+ *     copied to the new page.  \ref g_pntrTempAllocStackTop may be reset to
+ *     \ref g_pntrStartTempAllocStack.
+ *
+ * -# The submitted parameters are used to create a new line from their values.
+ *     Placeholders in \p fmt are replaced with their respective and to text
+ *     converted data values.  All stored in an internal buffer first.
```

Comment:
I added 2 links in the description of parameter fmt.  Are they misplaced?

Reply:
I saw that once I got to the end. I think it might be good to indicate this upfront, something like
```
\p fmt and the arguments are passed to `printf` to render the string into an internal buffer.
```

Reply:
added a for details... hint